lib/lita/handlers/artifactory.rb

```diff
@@ -126,6 +126,11 @@ def promote(response)
           EOH
         end
 
+        # This event allows other lita utilities to act when projects are promoted.
+        robot.trigger(:artifactory_artifact_promoted, product: project, version: version)
```

Comment:
Ahh you are correct, `source_channel` isn't returned in the response from:
https://github.com/chef-cookbooks/oc-artifactory/blob/master/files/default/plugins/Promotions.groovy

I suppose a simple map in that class could handle the source => target mapping....we know the promotion shape:

```ruby
def source_channel(target_channel)
  case target_channel
  when 'current'; 'unstable'
  when 'stable'; 'current'
  else raise "Source channel is not known for target channel: #{target_channel}"
  end
end```